src/string/src/lib.rs

```diff
@@ -12,13 +14,41 @@ use core::cmp;
 use core::usize;
 
 #[no_mangle]
-pub extern "C" fn memccpy(s1: *mut c_void, s2: *const c_void, c: c_int, n: usize) -> *mut c_void {
-    unimplemented!();
+pub unsafe extern "C" fn memccpy(
+    dest: *mut c_void,
+    src: *const c_void,
+    c: c_int,
+    n: usize,
+) -> *mut c_void {
+    use compiler_builtins::mem::memcpy;
+    use core::mem;
```

Comment:
This can be removed right?

Reply:
It's being used for `size_of` in line 32.

Reply:
In the revised commits, I have removed it and the `size_of` as `size_of::<u8>()` should always be 1. I can change it back.